src/applications/disability-benefits/526EZ/actions/index.js

```diff
@@ -1,64 +1,158 @@
-  expired: 'expired',
+import existingITFData from '../tests/itfData';
+
+export const PRESTART_STATUS_SET = 'PRESTART_STATUS_SET';
+export const PRESTART_DATA_SET = 'PRESTART_DATA_SET';
+export const PRESTART_STATE_RESET = 'PRESTART_STATE_RESET';
+export const PRESTART_DISPLAY_RESET = 'PRESTART_DISPLAY_RESET';
+
+export const PRESTART_STATUSES = {
+  notAttempted: 'not-attempted',
   pending: 'pending',
-  incomplete: 'incomplete'
-});
-
-
-// TODO: replace mock once ITF endpoint setup
-export function submitIntentToFile(formConfig, onChange) {
-  const { intentToFileUrl, formId, migrations, prefillTransformer } = formConfig;
-  let ITFStatus = ITFStatuses.pending;
-  return dispatch => {
-
-    onChange(ITFStatus);
-
-    const delay = (ms) => new Promise((resolve, reject) => {
-      ITFStatus = ITFStatuses.active;
-      onChange({ ITFStatus });
-
-      // TODO: if the backend handles resubmission, this check can be removed
-      if (ITFStatus === ITFStatuses.active) {
-        setTimeout(resolve, ms);
-      } else {
-        const errorMessage = 'Network request failed';
-        onChange({
-          ITFStatus,
-          errorMessage
-        });
-        reject(errorMessage);
-      }
-    });
-
-    return delay(2000);
+  succeeded: 'succeeded',
+  failed: 'failed'
+};
+
+export const PRESTART_VERIFICATION_TYPES = {
+  retrieve: 'retrieve',
+  create: 'create'
+};
+
+export const ITF_STATUSES = {
+  expired: 'expired',
+  active: 'active'
+};
+
+export function setPrestartStatus(status) {
+  return {
+    type: PRESTART_STATUS_SET,
+    status
+  };
+}
+
+export function setPrestartData(data) {
+  return {
+    type: PRESTART_DATA_SET,
+    data
   };
 }
 
-//   apiRequest(
-//     `${intentToFileUrl}`,
-//     null,
-//     ({ data }) => {
-//       const ITFStatus = data.attributes.ITFStatus
-//       onChange({ ITFStatus });
-//       return ITFStatus === 'active';
-//     },
-//     ({ errors }) => {
-//       const errorMessage = 'Network request failed';
-//       onChange({
-//         ITFStatus,
-//         errorMessage
-//       });
-//       Raven.captureMessage(`vets_itf_error: ${errorMessage}`);
-//     }
-//   );
-// };
-// }
-/* eslint-enable */
+export function resetPrestartState() {
+  return {
+    type: PRESTART_STATE_RESET
+  };
+}
+
+export function resetPrestartDisplay() {
+  return {
+    type: PRESTART_DISPLAY_RESET
+  };
+}
+
+// Returns most recent timestamp from a list of timestamps
+export const getLatestTimestamp = (timestamps) => timestamps.sort((a, b) => new Date(b) - new Date(a))[0];
+export const isNotEmptyObject = (object) => !(Object.keys(object).length === 0 && object.constructor === Object);
+export const getITFsByStatus = (itfs, status) => itfs.filter(itf => itf.status === status);
+
+export const handleCheckSuccess = (data, dispatch) => {
+  // If no existing ITF(s) are found, we will receive an empty {} response
+  const itfList = isNotEmptyObject(data) && data.attributes.intentToFile;
```

Comment:
`data` is deconstructed from the response in `checkITFRequest`. In order for this to work, the response would have to look like
```
{
  data: {}
}
```
When there are no ITFs, the response is going to look like
```
{}
```
This means `data`, when it gets here, will be undefined, not an empty object. We should be able to do `const itfList = get('attributes.intentToFile', data)`.

Reply:
Thanks for double checking this @cvalarida, @SMLuthi could you please explain what we will expect data to be when the response body is an empty object? Will the BE serialize this as an empty object as well? Or as undefined?

Reply:
Ah since we're using apiResponse, it won't use the `data` attribute on the response (which will be the serialized body) in this case. I wonder if we would want to update apiRequest to return the serialized data if it is already present, since that seems fairly standard. I'll update as you recommend.

Reply:
This is what a call to `/v0/intent_to_file` will look like if the user has no itfs and the response is `{}`

```
{
    "data": {
        "attributes": {
            "intent_to_file": []
        },
        "id": "",
        "type": "evss_intent_to_file_intent_to_files_responses"
    }
}
```

Reply:
Any response from vets-api that is serialized will follow the above format.

Reply:
Thanks so much @SMLuthi, will update accordingly.